libs/data/src/sort/sort.ts

```diff
@@ -2,26 +2,28 @@ import { IAuction } from '../auctions';
 import { SortDirection } from './sort-direction';
 import { SortKey } from './sort-key';
 
-export function sortAuctions(auctions: IAuction[], sortBy: SortKey, sortDir: SortDirection): IAuction[] {
-  const sortedAuctions = [...auctions];
+const compareString = (fst: string | Date, scd: string | Date) =>
+  `${fst}`.toLocaleLowerCase().localeCompare(`${scd}`.toLocaleLowerCase());
 
-  if (sortBy === SortKey.Heartcoins) {
-    sortDir === SortDirection.ASC
-      ? sortedAuctions.sort((a1, a2) => a1[sortBy] - a2[sortBy])
-      : sortedAuctions.sort((a1, a2) => a2[sortBy] - a1[sortBy]);
+const compareNumber = (a: number, b: number) => a - b;
 
-    return sortedAuctions;
-  }
+const withStrategy =
+  <Strategy extends SortKey, A extends IAuction>(
+    sortDir: boolean,
+    sortBy: Strategy,
+    sortStrategy: (a: A[Strategy], b: A[Strategy]) => number
+  ) =>
+  (auction1: A, auction2: A) => {
+    const [fst, scd] = sortDir ? [auction1, auction2] : [auction2, auction1];
```

Comment:
I like the new way the sorting looks. If I understand correctly fst is ASC sorting and scd is DESC sorting. If I am correct, it would be better to change the name that fits better. 
For example **fst** change to **asc**.

Reply:
Done